made floor/ceil return ints

Chillee opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #21131 Added degrees, radians, ldexp
* #21130 Added frexp, isinf, isnan, isfinite
* #21129 Added factorial
* #21128 Finished trigonometric functions
* #21127 Finished the high-priority functions
* #21126 Added remainder of high-priority trigonometric math ops
* #21125 Added better tests for math ops and unified them
* **#21124 made floor/ceil return ints**
* #21166 Added base parameter to math.log

Differential Revision: [D15563187](https://our.internmc.facebook.com/intern/diff/D15563187)

Chillee committed:
made floor/ceil return ints

Chillee committed:
Update on "made floor/ceil return ints"

made floor/ceil return ints

gh-metadata: pytorch pytorch 21124 gh/chillee/2/head

Chillee committed:
Update on "made floor/ceil return ints"

made floor/ceil return ints

gh-metadata: pytorch pytorch 21124 gh/chillee/2/head

Chillee committed:
Update on "made floor/ceil return ints"

made floor/ceil return ints

gh-metadata: pytorch pytorch 21124 gh/chillee/2/head

eellison commented on torch/csrc/jit/register_prim_ops.cpp:
```diff
@@ -96,6 +96,23 @@ static int64_t floordiv(int64_t a, int64_t b) {
+  if (std::isnan(a) || std::isinf(a) ||
+      a > double(std::numeric_limits<int64_t>::max()) ||
+      a < double(std::numeric_limits<int64_t>::min())) {
+    throw std::runtime_error(
```
Can you use our internal error `c10::Error`? We haven't been great about maintaining this, but in certain places (like constant propagation) we rely on  distinguishing errors thrown by us versus general c++ errors

Chillee replied:
It seems we actually don't use `c10::Error` anywhere, except for one instance in constant prop. All the other similar errors in this file are also using `std::runtime_error`, should we eventually switch all of them?

eellison replied:
Yea, we should eventually switch them. Every AT_ASSERT, AT_ERROR, etc uses `c10::Error` so it's not just the one place that we catch currently

Chillee replied:
Done.

Chillee committed:
Update on "made floor/ceil return ints"

made floor/ceil return ints

gh-metadata: pytorch pytorch 21124 gh/chillee/2/head

Chillee committed:
Update on "made floor/ceil return ints"

made floor/ceil return ints

gh-metadata: pytorch pytorch 21124 gh/chillee/2/head

Chillee committed:
Update on "made floor/ceil return ints"

made floor/ceil return ints

gh-metadata: pytorch pytorch 21124 gh/chillee/2/head

Chillee committed:
Update on "made floor/ceil return ints"

made floor/ceil return ints

gh-metadata: pytorch pytorch 21124 gh/chillee/2/head

Chillee committed:
Update on "made floor/ceil return ints"

made floor/ceil return ints

gh-metadata: pytorch pytorch 21124 gh/chillee/2/head

Chillee committed:
Update on "made floor/ceil return ints"

made floor/ceil return ints

gh-metadata: pytorch pytorch 21124 gh/chillee/2/head

ailzhang commented on torch/csrc/jit/register_prim_ops.cpp:
```diff
@@ -96,6 +96,23 @@ static int64_t floordiv(int64_t a, int64_t b) {
+  if (std::isnan(a) || std::isinf(a) ||
+      a > double(std::numeric_limits<int64_t>::max()) ||
+      a < double(std::numeric_limits<int64_t>::min())) {
+    throw c10::Error(
```
Do we have a test to trigger this error btw?

Chillee replied:
Not now. I don't think it's too necessary - our behavior when the user passes in an invalid input is 1. not really defined, and 2. more permissive than Python (other than with >64 bit integers). I think it's likely that any tests we write are likely to become code churn as various things change.

ailzhang reviewed (approved):
Thanks! Please land once CI is green.

Chillee commented:
It's actually green (just rate limited). Thanks!
